runtime/executor/src/main/java/org/apache/nemo/runtime/executor/datatransfer/BlockOutputWriter.java

```diff
@@ -74,21 +83,51 @@
       .orElseThrow(() -> new RuntimeException("No data store property on the edge"));
     blockToWrite = blockManagerWorker.createBlock(
       RuntimeIdManager.generateBlockId(runtimeEdge.getId(), srcTaskId), blockStoreValue);
+    potentialSpilledBlocktoWrite = blockManagerWorker.createBlock(
+      RuntimeIdManager.generateBlockId(runtimeEdge.getId(), srcTaskId), DataStoreProperty.Value.LOCAL_FILE_STORE);
+    LOG.info("dongjoo, BlockOutputWriter constructor, stageEdge {}, runtimeEdge {}, blockToWrite {}",
+      stageEdge, runtimeEdge, blockToWrite);
     final Optional<DuplicateEdgeGroupPropertyValue> duplicateDataProperty =
       runtimeEdge.getPropertyValue(DuplicateEdgeGroupProperty.class);
     nonDummyBlock = !duplicateDataProperty.isPresent()
       || duplicateDataProperty.get().getRepresentativeEdgeId().equals(runtimeEdge.getId())
       || duplicateDataProperty.get().getGroupSize() <= 1;
+
+    //dongjoo
+    this.memoryManager = memoryManager;
+    this.sizeTrackingVector = new SizeTrackingVector();
+
   }
 
+
   @Override
   public void write(final Object element) {
+//    LOG.info("type of partitioner, key: {} and type of block {}", partitioner.partition(element), blockStoreValue);
     if (nonDummyBlock) {
-      blockToWrite.write(partitioner.partition(element), element);
-
+      // logging
+//      LOG.info("BlockOutPutWriter write, blockid {},blocktoWrite {}, blockStoreValue {}",
+//        blockToWrite.getId(), blockToWrite, blockStoreValue);
+//      LOG.info("written element {}", element);
+      // loggin common end
+      // if blockStore is caching to Memory, write to temporary SizeTrackingVector to ensure OOM doesn't occur,
+      // actual "writing" will occur in { close }
+      if (blockStoreValue == DataStoreProperty.Value.MEMORY_STORE
+        || blockStoreValue == DataStoreProperty.Value.SERIALIZED_MEMORY_STORE) {
+//        LOG.info("only for memory store append to STV");
+        sizeTrackingVector.append(element);
```

Comment:
How about maintaining a map of `BlockId`(or key of a Partition?)-`sizeTrackingVector` instead of reusing a single `sizeTrackingVector` for arbitrary Blocks?